src/core/api.c

```diff
@@ -631,8 +632,11 @@ MsQuicStreamOpen(
 
     QUIC_CONN_VERIFY(Connection, !Connection->State.Freed);
 
-    if (QuicConnIsClosed(Connection)) {
-        Status = QUIC_STATUS_INVALID_STATE;
+    if (QuicConnIsClosedLocation(Connection, &ClosedLocally)) {
```

Comment:
This PR is way over complicated. `QuicConnIsClosed` is just a wrapper checking two booleans. Just check them independently and return the error accordingly. No need for extra functions.